packages/core/crates/core-crypto/src/utils.rs

```diff
@@ -0,0 +1,20 @@
+/// Convenience method to XOR one slice onto other.
+pub fn xor_inplace(a: &mut [u8], b: &[u8]) {
+    let bound = if a.len() > b.len() { b.len() } else { a.len() };
+
+    // TODO: use portable_simd here
+    for i in 0..bound {
+        a[i] ^= b[i];
+    }
+}
+
+/// Convenience function to efficiently copy slices of unequal sizes.
+#[allow(dead_code)]
+pub fn copy_nonequal(target: &mut [u8], source: &[u8]) {
+    let sz = if target.len() > source.len() {
```

Comment:
Would `std::cmp::min` be more readable?

Reply:
Wanted to avoid using `std` for this simple thing.

Reply:
Yiu can use `Ord` then to make it more readable.